### PR TITLE
fix(gicd): fix calculation of gic's number of irqs

### PR DESCRIPTION
**Pull request by xioliu**
According to the specification, if the value of this fieldis N, the maximum SPI INTID is 32(N+1)-1.

### CODE REVIEW

**Review by danielRep — APPROVED** [review]
By reading the Arm GIC Architecture Specification v3 & v4, page 616

**ITLinesNumber, bits [4:0]:**
>For the INTID range 32 to 1019, indicates the maximum SPI supported.
If the value of this field is N, the maximum SPI INTID is 32(N+1) minus 1. For example, 00011 specifies that the maximum SPI INTID in is 127.

Please change the commit message to follow the conventional commit spec. My suggestion:
`fix(gicd): fix calculation of gic's number of irqs`

**Comment by xioliu** [comment]
I changed the commit message as your suggestion, please review, thank you.

**Comment by danielRep** [comment]
> I changed the commit message as your suggestion, please review, thank you.

Please squash the two commits @xioliu , you accidentally created an empty commit.

**Comment by xioliu** [comment]
Hello Daniel,

When I updated the commit message, github created the 2nd commit. I am not sure how I can squash the two commits as I don't see any buttons or icons for this operation. Please give me a clue, thank you.
image

**Comment by danielRep** [comment]
> Hello Daniel,
> 
> When I updated the commit message, github created the 2nd commit. I am not sure how I can squash the two commits as I don't see any buttons or icons for this operation. Please give me a clue, thank you. image

Sorry for the time to reply. You just need to use git rebase to perform a squash 
`git rebase -i HEAD~2` 
Follow the information from the command or just take a quick search, you will find N entries explaining how to do it.

**Comment by xioliu** [comment]
Hello Daniel,

Please review again.

**Comment by danielRep** [comment]
You squashed the commits but left the previous commit message. Amend it to `fix(gicd): fix calculation of gic's number of irqs`

**Comment by danielRep** [comment]
> Right, I forgot that. Done.

Sorry for the pedantic requests @xioliu , but you need to split the commit body message to avoid the gitlint checker throwing this error: 

`B1 Line exceeds max length (99>80): "According to the specification, if the value of this fieldis N, the maximum SPI INTID is 32(N+1)-1."`

**Comment by xioliu** [comment]
Hopefully it can pass now.

**Comment by danielRep** [comment]
> Hopefully it can pass now.

It's finish! Thanks for the contribution @xioliu

**Comment by xioliu** [comment]
Thank you for your support.

File: src/arch/armv8/inc/arch/gic.h
Comment (DavidMCerdeira) [inline src/arch/armv8/inc/arch/gic.h:441]: Although the original operation is indeed wrong, the correct formula is 32 * (N + 1).
Which should be:
```suggestion
    return 32 * (itlinenumber - 1);
```

**Review by DavidMCerdeira — CHANGES_REQUESTED** [review]

File: src/arch/armv8/inc/arch/gic.h
Comment (danielRep) [reply src/arch/armv8/inc/arch/gic.h:441]: Yes, I agree with this. Check page 89 of GICv2 specification.

**Review by danielRep — CHANGES_REQUESTED** [review]
Follow's @DavidMCerdeira suggested change @xioliu

File: src/arch/armv8/inc/arch/gic.h
Comment (DavidMCerdeira) [reply src/arch/armv8/inc/arch/gic.h:441]: Note that this function returns the number of interrupts, and not the range.

**Review by danielRep — APPROVED** [review]
Thanks @xioliu !

**Comment by DavidMCerdeira** [comment]
That's it.

Now please squash de two commits.
After this, it is ready to merge.

**Comment by xioliu** [comment]
Err, I screwed it up.

**Comment by xioliu** [comment]
I hope it works now. Sorry, I am a bad git user.

**Comment by DavidMCerdeira** [comment]
No worries @xioliu. However, there's one last issue... 

I assume you used the Github UI to add my code suggestion, which marked me as co-author. From my understanding, this requires that I should signoff the commit as well.
Given that the modification is trivial there's no need for me to be co-author. Please remove the "Co-authored-by: ..." from the commit message.

Sorry for such complexity in such a simple change. We are just trying to make sure everything is done correctly.

**Review by DavidMCerdeira — APPROVED** [review]

**Review by danielRep — APPROVED** [review]

**Pull request merged**
